src/zhinst/toolkit/driver/nodes/awg.py

```diff
@@ -167,47 +170,60 @@ def load_sequencer_program(
             )
 
     def write_to_waveform_memory(
-        self, waveforms: Waveforms, indexes: list = None
+        self, waveforms: Waveforms, indexes: list = None, check: bool = True
     ) -> None:
         """Writes waveforms to the waveform memory.
 
         The waveforms must already be assigned in the sequencer program.
 
         Args:
             waveforms: Waveforms that should be uploaded.
+            indexes: Specify a list of indexes that should be uploaded. If
+                nothing is specified all available indexes in waveforms will
+                be uploaded. (default = None)
+            check: Enable sanity check preformed by toolkit, based on the
```

Comment:
Would `validate` be better argument name? As Toolkit validates the  waveforms.

Reply:
good point